src/server/journal/journal_slice.cc

```diff
@@ -159,5 +164,18 @@ void JournalSlice::UnregisterOnChange(uint32_t id) {
   change_cb_arr_.erase(it);
 }
 
+void JournalSlice::PingsLoop(std::chrono::milliseconds period_ms) {
+  Entry entry(/*txid=*/0, Op::PING, /*dbid=*/0,
+              /*shard_cnt=*/1, /*pl=*/{});
+  while (!lameduck_) {
+    VLOG(1) << "Sending PING to journal writer.";
+    AddLogRecord(entry, true);
```

Comment:
Note that if you allow preemption when adding to log (second arg is true)
than this pings might not be sent every period_ms